node/test/toolrunnertests.ts

```diff
@@ -395,8 +395,68 @@ describe('Toolrunner Tests', function () {
                 done(err);
             });
     })
+    // function to compile a .NET program that prints lines.
+    // the helper program is used on Windows to validate piping output between tools.
+    let compileOutputExe = (): string => {
```

Comment:
there is a lot of duplication here. i'll refactor the common code into a separate function.